amundsen_application/static/js/components/TableDetail/EditableSection/index.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+
+import './styles.scss';
+
+export interface EditableSectionProps {
+  title: string;
+}
+
+interface EditableSectionState {
+  isEditing: boolean;
+}
+
+export interface EditableSectionChildProps {
+  isEditing?: boolean;
+  setEditMode?: (isEditing: boolean) => void;
+}
+
+export class EditableSection extends React.Component<EditableSectionProps, EditableSectionState> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isEditing: false,
+    }
+  }
+
+  setEditMode = (isEditing: boolean) => {
+    this.setState({ isEditing });
+  };
+
+  toggleEdit = () => {
+    this.setState({ isEditing: !this.state.isEditing });
+  };
+
+  render() {
+    const childrenWithProps = React.Children.map(this.props.children, child => {
+      if (!React.isValidElement(child)) {
+        return child;
```

Comment:
nit: I think you could cover this case by using a setup with some plain html markup `const element = (<div>Hi</div>);` and then verifying that the render() result is that element? Not blocking merge on this though.

Reply:
Added a case for when the child is just a `string`. Using `<div>` actually counts as a `React.isValidElement` and will attempt to put the props on the div, which gives further warnings.

Reply:
If that's the case, is `React.isValidElement` the right method to use then? Do we at least get those warnings when the code tries to build?

I'd want to make sure we avoid the possibility for someone to use `EditableSection` incorrectly, yet still end up with a successful build and deploy and then the page breaks in production.

Reply:
I'll do some research on type-checking children and modify this as a follow up commit.